src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -311,8 +311,12 @@ private void applyNewIndices(final ClusterChangedEvent event) {
                 }
                 try {
                     indicesService.createIndex(indexMetaData.index(), indexMetaData.settings(), event.state().nodes().localNode().id());
-                } catch (Exception e) {
-                    logger.warn("[{}] failed to create index", e, indexMetaData.index());
+                } catch (Throwable e) {
+                    logger.warn("[{}] failed to create index for shard {}", e, indexMetaData.index(), shard.shardId());
```

Comment:
does it make sense to put the "fail a shard" logic under a method? to make sure we don't forget to put it in failedShards etc.